external-crates/move/crates/move-package/src/migration/mod.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::BTreeMap,
+    io::{BufRead, Write},
+};
+
+use move_command_line_common::interactive::Terminal;
+use move_compiler::{diagnostics::Migration, editions::Edition};
+use once_cell::sync::Lazy;
+
+use crate::compilation::build_plan::BuildPlan;
+
+pub const MIGRATION_MSG: &str =
+    "Package toml does not specify an edition. As of 2024, Move requires all packages to define \
+    a language edition.";
+
+pub const EDITION_SELECT_PROMPT: &str = "Please select one of the following editions:";
+
+pub static EDITION_OPTIONS: Lazy<BTreeMap<String, Edition>> = Lazy::new(|| {
+    let mut map = BTreeMap::new();
+    map.insert("1".to_string(), Edition::E2024_ALPHA);
+    map.insert("3".to_string(), Edition::LEGACY);
+    map
+});
+
+pub const EDITION_RECORDED_MSG: &str = "Recorded edition in Move.toml";
+
+pub const MIGRATION_PROMPT: &str =
+    "Would you like the Move compiler to migrate your code to Move 2024?";
+
+pub const NOMIGRATION_HELP_MSG: &str = "No migration was performed.";
+
+pub const MIGRATION_RERUN: &str = "You can rerun this migration by calling 'move migrate' \
+    or removing the edition from your package's 'Move.toml' file.";
```

Comment:
Is there something I need to do after I remove the edition from the package?

Reply:
You have to re-call build.